scripts/analysis_1/nbconverted/3_visualize_data.py

```diff
@@ -0,0 +1,236 @@
+
+weights_decoder_file = glob.glob(os.path.join(
+    NN_dir,
+    "tybalt_2layer_{}latent_decoder_weights.h5".format(latent_dim)))[0]
+
+experiment_dir = os.path.join(
+    local_dir,
+    "Data",
+    "Batch_effects",
+    "experiment_simulated",
+    analysis_name)
+
+simulated_data_file = os.path.join(
+    local_dir,
+    "Data",
+    "Batch_effects",
+    "simulated",
+    analysis_name,
+    "simulated_data.txt.xz")
+
+permuted_simulated_data_file = os.path.join(
+    local_dir,
+    "Data",
+    "Batch_effects",
+    "simulated",
+    analysis_name,
+    "permuted_simulated_data.txt.xz")
+
+
+# ## 1. Visualize simulated data in UMAP space
+
+# In[4]:
+
+
+# Load saved models
+loaded_model = load_model(model_encoder_file)
+loaded_decode_model = load_model(model_decoder_file)
+
+loaded_model.load_weights(weights_encoder_file)
+loaded_decode_model.load_weights(weights_decoder_file)
+
+
+# In[5]:
+
+
+# Read data
+normalized_data = pd.read_table(
+    normalized_data_file,
+    header=0,
+    sep='\t',
+    index_col=0).T
+
+simulated_data = pd.read_table(
+    simulated_data_file,
+    header=0,
+    sep='\t',
+    index_col=0)
+
+print(normalized_data.shape)
+print(simulated_data.shape)
+
+
+# In[6]:
+
+
+normalized_data.head(10)
+
+
+# In[7]:
+
+
+simulated_data.head(10)
+
+
+# In[8]:
+
+
+# Add labels to original normalized data
+sample_ids = list(simulated_data.index)
+
+
+normalized_data_label = normalized_data.copy()
+
+normalized_data_label['color_by'] = 'XXX'
+normalized_data_label.loc[sample_ids, 'color_by'] = simulated_data['color_by']
+normalized_data_label.loc[sample_ids].head(10)
+
+
+# In[13]:
+
+
+# UMAP embedding of original input data
+
+# Get and save model
+model = umap.UMAP(random_state=randomState).fit(normalized_data)
+
+input_data_UMAPencoded = model.transform(normalized_data)
+input_data_UMAPencoded_df = pd.DataFrame(data=input_data_UMAPencoded,
+                                         index=normalized_data.index,
+                                         columns=['1','2'])
+# Add label
+input_data_UMAPencoded_df['color_by'] = normalized_data_label['color_by']
+
+ggplot(input_data_UMAPencoded_df, aes(x='1',y='2'))     + geom_point(aes(color='color_by'), alpha=1)     + xlim(7, 11)     + ylim(-3, -2.5)     + ggtitle('Input data')
+
+
+# In[10]:
+
+
+# UMAP embedding of simulated data
+
+# Drop label column
+simulated_data_numeric = simulated_data.drop(['color_by'], axis=1)
+
+simulated_data_UMAPencoded = model.transform(simulated_data_numeric)
+simulated_data_UMAPencoded_df = pd.DataFrame(data=simulated_data_UMAPencoded,
+                                         index=simulated_data.index,
+                                         columns=['1','2'])
+
+# Add back label column
+simulated_data_UMAPencoded_df['color_by'] = simulated_data['color_by']
+
+
+ggplot(simulated_data_UMAPencoded_df, aes(x='1',y='2'))     + geom_point(aes(color='color_by'), alpha=1)     + ggtitle("Simulated data")
```

Comment:
color to label inconsistent from above fig

Reply:
I would really like the plots to have the same color.  For now I have figured out how to translate color palettes between two figures.  Since these figures are just to visualize the data as a means to debug and so the figures (and therefore coloring requirements) will change moving forward, I will leave the coloring for now since I couldn't find a quick solution.